app/components/PlatformCards/SnapshotPlatform.tsx

```diff
@@ -0,0 +1,151 @@
+
+  // SelectedProviders will be passed in to the sidebar to be filled there...
+  const [verifiedProviders, setVerifiedProviders] = useState<PROVIDER_ID[]>(
+    providerIds.filter((providerId) => typeof allProvidersState[providerId]?.stamp?.credential !== "undefined")
+  );
+  // SelectedProviders will be passed in to the sidebar to be filled there...
+  const [selectedProviders, setSelectedProviders] = useState<PROVIDER_ID[]>([...verifiedProviders]);
+
+  // any time we change selection state...
+  useEffect(() => {
+    if (selectedProviders.length !== verifiedProviders.length) {
+      setCanSubmit(true);
+    }
+    if (selectedProviders.length === 0) {
+      setCanSubmit(false);
+    }
+  }, [selectedProviders, verifiedProviders]);
+
+  // --- Chakra functions
+  const toast = useToast();
+
+  // // if the selected providers length equals 0, disable the Verify button
+  // useEffect(() => {
+  //   if (selectedProviders.length === 0) {
+  //     setCanSubmit(false);
+  //   }
+  // }, [selectedProviders]);
+
+  // fetch VCs from IAM server
+  const handleFetchCredential = (): void => {
+    datadogLogs.logger.info("Saving Stamp", { platform: platformId });
+    fetchVerifiableCredential(
+      iamUrl,
+      {
+        type: platformId,
+        types: selectedProviders,
+        version: "0.0.0",
+        address: address || "",
+        proofs: {},
+      },
+      signer as { signMessage: (message: string) => Promise<string> }
+    )
+      .then(async (verified: VerifiableCredentialRecord): Promise<void> => {
+        // because we provided a types array in the params we expect to receive a
+        // credentials array in the response...
+        const vcs =
+          verified.credentials
+            ?.map((cred: CredentialResponseBody): Stamp | undefined => {
+              if (!cred.error) {
+                // add each of the requested/received stamps to the passport...
+                return {
+                  provider: cred.record?.type as PROVIDER_ID,
+                  credential: cred.credential as VerifiableCredential,
+                };
+              }
+            })
+            .filter((v: Stamp | undefined) => v) || [];
+        // Add all the stamps to the passport at once
+        await handleAddStamps(vcs as Stamp[]);
+        datadogLogs.logger.info("Successfully saved Stamp", { platform: platformId });
+        const verifiedProviders = providerIds.filter(
+          (providerId) => typeof allProvidersState[providerId]?.stamp?.credential !== "undefined"
+        );
+        // update the verified and selected providers
+        setVerifiedProviders([...verifiedProviders]);
+        setSelectedProviders([...verifiedProviders]);
+        // reset can submit state
+        setCanSubmit(false);
+        // Custom Success Toast
+        toast({
+          duration: 5000,
+          isClosable: true,
+          render: (result: any) => <DoneToastContent providerId={platformId} result={result} />,
```

Comment:
We will need to make the same `providerId` -> `platformId` change here too:

```suggestion
          render: (result: any) => <DoneToastContent platformId={platformId} result={result} />,
```